File: src/main/templates/typescript_sdk/src/client/FlexmodelClientFactory.ts
import { ApiClient, ApiClientOptions } from './ApiClient.js';
import { FlexmodelClient } from './FlexmodelClient.js';

/**
 * Flexmodel客户端工厂
 * 提供便捷的客户端创建方法
 */
export class FlexmodelClientFactory {
  /**
   * 创建Flexmodel客户端
   * @param baseURL 基础URL
   * @param datasourceName 数据源名称
   * @param options 选项
   * @returns Flexmodel客户端
   */
  static createClient(baseURL: string, datasourceName: string, options: ApiClientOptions = {}): FlexmodelClient {
    const apiClient = new ApiClient(baseURL, options);
    return new FlexmodelClient(apiClient, datasourceName);
  }

  /**
   * 创建Flexmodel客户端（带API Key认证）
   * @param baseURL 基础URL
   * @param datasourceName 数据源名称
   * @param apiKey API密钥
   * @param options 其他选项
   * @returns Flexmodel客户端
   */
  static createClientWithApiKey(
    baseURL: string, 
    datasourceName: string, 
    apiKey: string, 
    options: ApiClientOptions = {}
  ): FlexmodelClient {
    return this.createClient(baseURL, datasourceName, {
      ...options,
      apiKey
    });
  }

  /**
   * 创建Flexmodel客户端（带用户名密码认证）
   * @param baseURL 基础URL
   * @param datasourceName 数据源名称
   * @param username 用户名
   * @param password 密码
   * @param options 其他选项
   * @returns Flexmodel客户端
   */
  static createClientWithCredentials(
    baseURL: string, 
    datasourceName: string, 
    username: string, 
    password: string, 
    options: ApiClientOptions = {}
  ): FlexmodelClient {
    return this.createClient(baseURL, datasourceName, {
      ...options,
      username,
      password
    });
  }

  /**
   * 创建默认配置的客户端（用于开发环境）
   * @param datasourceName 数据源名称
   * @param options 其他选项
   * @returns Flexmodel客户端
   */
  static createDefaultClient(datasourceName: string, options: ApiClientOptions = {}): FlexmodelClient {
    return this.createClient('http://localhost:8080', datasourceName, options);
  }
}

File: src/main/templates/java_sdk/src/main/java/${path}/client/FlexmodelClientFactory.java
package ${packageName};

/**
 * Flexmodel客户端工厂
 * 提供便捷的客户端创建方法
 */
public class FlexmodelClientFactory {

    /**
     * 创建Flexmodel客户端
     * @param baseUrl 基础URL
     * @param datasourceName 数据源名称
     * @return Flexmodel客户端
     */
    public static FlexmodelClient createClient(String baseUrl, String datasourceName) {
        ApiClient apiClient = new ApiClient(baseUrl);
        return new FlexmodelClient(apiClient, datasourceName);
    }

    /**
     * 创建Flexmodel客户端（带认证）
     * @param baseUrl 基础URL
     * @param datasourceName 数据源名称
     * @param apiKey API密钥
     * @return Flexmodel客户端
     */
    public static FlexmodelClient createClient(String baseUrl, String datasourceName, String apiKey) {
        ApiClient apiClient = new ApiClient(baseUrl);
        apiClient.setApiKey(apiKey);
        return new FlexmodelClient(apiClient, datasourceName);
    }

    /**
     * 创建Flexmodel客户端（带用户名密码认证）
     * @param baseUrl 基础URL
     * @param datasourceName 数据源名称
     * @param username 用户名
     * @param password 密码
     * @return Flexmodel客户端
     */
    public static FlexmodelClient createClient(String baseUrl, String datasourceName, String username, String password) {
        ApiClient apiClient = new ApiClient(baseUrl);
        apiClient.setUsername(username);
        apiClient.setPassword(password);
        return new FlexmodelClient(apiClient, datasourceName);
    }

    /**
     * 创建默认配置的客户端（用于开发环境）
     * @param datasourceName 数据源名称
     * @return Flexmodel客户端
     */
    public static FlexmodelClient createDefaultClient(String datasourceName) {
        return createClient("http://localhost:8080", datasourceName);
    }
}
